Add documentation for customizing the webhook. (#1081)
rancher/rancher-docs: docs/reference-guides/rancher-webhook.md

kubectl create -f example.yaml --as=system:serviceaccount:cattle-system:rancher-webhook-sudo --as-group=system:masters
```

## Customizing Rancher-Webhook Configuration
Rancher allows users to add custom helm values to the helm installation of Rancher-Webhook. When Rancher installs the rancher-webhook chart via helm it will check for custom helm values defined in a ConfigMap with the name `rancher-config` in the `cattle-system` namespace under the data key `rancher-webhook`. The value of this key must be valid YAML.
``` yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: rancher-config
  namespace: cattle-system
  labels:
    app.kubernetes.io/part-of: "rancher"
data:
  rancher-webhook: '{"port": 9553, "priorityClassName": "system-node-critical"}'

```
Rancher will redeploy the Rancher-Webhook chart when changes to the ConfigMap values are detected.

> ### Customising Rancher-Webhook During Rancher Installation

Reviewer: ```suggestion
### Customizing Rancher-Webhook During Rancher Installation
```

> You can customize the local cluster's Rancher-webhook's helm values during the initial Helm installation of the Rancher chart. All values used in the Rancher-Webhook chart can be accessed as nested variables under the `webhook` name. 

Reviewer 2: ```suggestion
When you use Helm to install the Rancher chart, you can add custom Helm values to the Rancher-Webhook of the local cluster. All values in the Rancher-Webhook chart are accessible as nested variables under the `webhook` name.
```

> These values will be synced to the `rancher-config` ConfigMap during installation.

Reviewer 2: ```suggestion
These values are synced to the `rancher-config` ConfigMap during installation.
```

```bash
helm install rancher rancher-<CHART_REPO>/rancher \
  --namespace cattle-system \
  ... 
  --set webhook.port=9553 \
  --set webhook.priorityClassName="system-node-critical"
```

## Common Issues

### EKS Cluster with Calico CNI

Users running an EKS cluster with Calico CNI may run into errors when the Kubernetes API server attempts to contact the Rancher-Webhook. 

> One workaround for this issue [documented by Calico](https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks#install-eks-with-calico-networking) involves setting `hostNetwork=true` for the webhook deployment. Users can change this by adding the helm value `global.hostNetwork=true` to the value to the `rancher-config` ConfigMap see [Customizing Rancher-Webhook Configuration](#customizing-rancher-webhook-configuration) for more info.

Reviewer: ```suggestion
One workaround for this issue [documented by Calico](https://docs.tigera.io/calico/latest/getting-started/kubernetes/managed-public-cloud/eks#install-eks-with-calico-networking) involves setting `hostNetwork=true` for the webhook deployment. Users can change this by adding the helm value `global.hostNetwork=true` to the `rancher-config` ConfigMap see [Customizing Rancher-Webhook Configuration](#customizing-rancher-webhook-configuration) for more info.
```

``` bash
apiVersion: v1
kind: ConfigMap
metadata:
  name: rancher-config
  namespace: cattle-system
  labels:
    app.kubernetes.io/part-of: "rancher"
data:
  rancher-webhook: '{"global": {"hostNetwork": true}}'
```

**Note:** This temporary workaround may violate an environment's security policy. This workaround also requires that port 9443 is unused on the host network.

> **Note:** Helm uses secrets by default to store information. This is a resource that some webhook versions validate. In these cases, directly update the deployment with the hostNetwork=true value using kubectl, then update the webhook configuration as specified above.

Reviewer 2: ```suggestion
**Note:** By default, Helm stores information as secrets. Secrets are a resource that some webhook versions validate. In these cases, directly update the deployment with the `hostNetwork=true` value using kubectl, then update the webhook configuration as specified above.
```
